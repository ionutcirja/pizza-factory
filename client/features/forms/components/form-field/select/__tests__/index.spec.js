import React from 'react';
import { shallow } from 'enzyme';
import { SelectField } from '..';

describe('SelectField component', () => {
  let propsToRender;
  
  beforeEach(() => {
    propsToRender = {
      field: {
        name: 'select',
        value: '',
      },
      form: {
        touched: {},
        errors: {},
      },
      options: [
        'option1',
        'option2',
        'option3',
      ],
      label: 'label',
      theme: {
        colours: {
          grey: 'grey',
          black: 'black',
          red: 'red',
        },
      },
    };
  });
  
  describe('render', () => {
    it('should render a select field containing a list of options', () => {
      const wrapper = shallow(<SelectField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should render a disabled select field if disabled prop value is truthy', () => {
      propsToRender.disabled = true;
      const wrapper = shallow(<SelectField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render a disabled default option if placeholder prop value is truthy', () => {
      propsToRender.placeholder = 'select an option';
      const wrapper = shallow(<SelectField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should render an error if field is touched and'
      + ' there is an error defined for the current field', () => {
      propsToRender.form = {
        touched: {
          select: true,
        },
        errors: {
          select: 'some error',
        },
      };
      const wrapper = shallow(<SelectField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
