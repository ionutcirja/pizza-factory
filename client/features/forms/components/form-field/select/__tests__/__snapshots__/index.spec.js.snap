// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField component render should render a disabled default option if placeholder prop value is truthy 1`] = `
<FieldSet>
  <Select
    name="select"
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      select an option
    </option>
    <option
      key="option1"
      value="option1"
    >
      option1
    </option>
    <option
      key="option2"
      value="option2"
    >
      option2
    </option>
    <option
      key="option3"
      value="option3"
    >
      option3
    </option>
  </Select>
</FieldSet>
`;

exports[`SelectField component render should render a disabled select field if disabled prop value is truthy 1`] = `
<FieldSet>
  <Select
    disabled={true}
    name="select"
    value=""
  >
    <option
      key="option1"
      value="option1"
    >
      option1
    </option>
    <option
      key="option2"
      value="option2"
    >
      option2
    </option>
    <option
      key="option3"
      value="option3"
    >
      option3
    </option>
  </Select>
</FieldSet>
`;

exports[`SelectField component render should render a select field containing a list of options 1`] = `
<FieldSet>
  <Select
    name="select"
    value=""
  >
    <option
      key="option1"
      value="option1"
    >
      option1
    </option>
    <option
      key="option2"
      value="option2"
    >
      option2
    </option>
    <option
      key="option3"
      value="option3"
    >
      option3
    </option>
  </Select>
</FieldSet>
`;

exports[`SelectField component render should render an error if field is touched and there is an error defined for the current field 1`] = `
<FieldSet>
  <Select
    name="select"
    value=""
  >
    <option
      key="option1"
      value="option1"
    >
      option1
    </option>
    <option
      key="option2"
      value="option2"
    >
      option2
    </option>
    <option
      key="option3"
      value="option3"
    >
      option3
    </option>
  </Select>
  <Error>
    some error
  </Error>
</FieldSet>
`;
